Core/src/AmbiguityResolution/ScoreBasedAmbiguityResolution.cpp

```diff
@@ -7,3 +7,19 @@
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
 #include "Acts/AmbiguityResolution/ScoreBasedAmbiguityResolution.hpp"
+
+bool Acts::ScoreBasedAmbiguityResolution::etaBasedCuts(
+    const DetectorConfig& detector, const TrackFeatures& trackFeatures,
+    const double& eta) const {
+  std::vector<double> etaBins = detector.etaBins;
+
+  auto it = std::upper_bound(etaBins.begin(), etaBins.end(), eta);
```

Comment:
```suggestion
  auto it = std::ranges::upper_bound(etaBins, eta);
```

Though, I am not sure, if our compiler does already support this range feature.